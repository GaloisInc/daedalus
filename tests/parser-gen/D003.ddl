-- test overlap of intervals

def Main =
  { $$ = Choose
    { x = { $['0'..'7'] ; $['a']; }
    ; y = { $['3'..'9'] ; $['b']; }
    }
  ; END
  }
