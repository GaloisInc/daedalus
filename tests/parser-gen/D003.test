dump
D003.ddl
--gen
