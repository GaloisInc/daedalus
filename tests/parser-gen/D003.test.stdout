Stack:
    *
    Main callsite __START__
Start: Main (5,33)-(5,43)
SlkCfg{ q:42, ctrl:[?, *], sem:[?, ?, ?, ], inp:--- }
DTrans [
  ( [(17),]
  , HeadInput ['a']
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
    Main callsite __START__
Start: Main (6,33)-(6,43)
SlkCfg{ q:18, ctrl:[?, *], sem:[?, ?, ?, ], inp:--- }
DTrans [
  ( [(17),]
  , HeadInput ['b']
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
    Main callsite __START__
Start: Main (8,5)-(8,7)
SlkCfg{ q:58, ctrl:[?, *], sem:[?, ], inp:--- }
DTrans [
  ( [(9),]
  , EndInput
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
Start: Main (3,1)-(9,3)
SlkCfg{ q:1, ctrl:[*], sem:[*], inp:--- }
DTrans [
  ( [(17),]
  , HeadInput ['0','2']
  , Resolution (NotAmbiguous)
  ),
  ( [(17),(17),]
  , HeadInput ['3','7']
  , DTrans [
      ( [(17),]
      , HeadInput ['a']
      , Resolution (NotAmbiguous)
      ),
      ( [(17),]
      , HeadInput ['b']
      , Resolution (NotAmbiguous)
      ),
    ]
  ),
  ( [(17),]
  , HeadInput ['8','9']
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
Start: __START__
SlkCfg{ q:0, ctrl:[*], sem:[?, *], inp:--- }
DTrans [
  ( [(0),]
  , AcceptingPath
  , NotAmbiguous
  ),
]

**********************
***** Strict LLA *****
**********************
("Lookahead-0",1)
("Lookahead-1",3)
("Lookahead-2",1)

Total nb DFAs: 5

SUCCESS: Fully determinized format
